src/engine/framework/BaseCommands.cpp

```diff
@@ -650,10 +653,45 @@ namespace Cmd {
                 const std::string& name = isNamed ? args.Argv(1) : "";
                 const std::string& command = args.EscapedArgs(2 + +isNamed);
                 std::string delay = args.Argv(1 + +isNamed);
-                bool frames = delay.back() == 'f';
 
-                if (frames) {
-                    delay.erase(--delay.end()); //FIXME-gcc-4.6 delay.pop_back()
+                const char front = !!delay.size() ? delay.front() : '\0';
+
+                if (front < '0' || front > '9') {
+                    Print("delay: missing delay");
+                    return;
+                }
+
+                enum class delayUnit {
+                    MILLISECOND = 0,
+                    SECOND = 's',
+                    MINUTE = 'm',
+                    HOUR = 'h',
+                    FRAME = 'f',
+                };
+
+                delayUnit unit = delayUnit::MILLISECOND;
+                char back = !!delay.size() ? delay.back() : '\0';
+
+                switch(delayUnit(back)) {
```

Comment:
I find this more readable, otherwise I'm looking everywhere where the function "delayUnit" is declared
```suggestion
                switch(static_cast<delayUnit>(back)) {
```